File: Wiwa/src/Wiwa/ecs/systems/game/player/PlayerSpawner.cpp
#include <wipch.h>
#include "PlayerSpawner.h"

namespace Wiwa
{
	Wiwa::PlayerSpawner::PlayerSpawner()
	{
	}

	PlayerSpawner::~PlayerSpawner()
	{
	}

	void PlayerSpawner::OnInit()
	{
		Transform3D* tr = GetComponent<Transform3D>();
		Wiwa::EntityManager& em = m_Scene->GetEntityManager();

		EntityId playerId = em.LoadPrefab("assets\\Player\\Final\\Player.wiprefab");
		Transform3D* playerTr = em.GetComponent<Transform3D>(playerId);

		playerTr->localPosition.x = tr->worldMatrix[3].x;
		playerTr->localPosition.y = tr->worldMatrix[3].y;
		playerTr->localPosition.z = tr->worldMatrix[3].z;

		playerTr->position.x = tr->worldMatrix[3].x;
		playerTr->position.y = tr->worldMatrix[3].y;
		playerTr->position.z = tr->worldMatrix[3].z;
	}
}

File: Editor/assets/Scripts/Player/PlayerSpawner.cs
//using System;
//using Wiwa;

//namespace Game
//{
//    using EntityId = UInt64;

//    class PlayerSpawner : Behaviour
//    {

//        void Init()
//        {
//            ref Transform3D tr = ref GetComponent<Transform3D>();

//            EntityId playerId = LoadPrefab("assets\\Player\\Final\\Player.wiprefab");
//            ref Transform3D playerTr = ref GetComponent<Transform3D>(playerId);

//            playerTr.LocalPosition = tr.worldMatrix.GetPosition();
//        }
//    }
//}
